alchemy/src/cloudflare/tanstack-start/plugin.ts

```diff
@@ -9,22 +10,41 @@ import { dedent } from "../../util/dedent.ts";
  * @see https://developers.cloudflare.com/workers/framework-guides/web-apps/tanstack/#using-cloudflare-bindings
  */
 export default function alchemy(options: GetPlatformProxyOptions = {}) {
-  return {
-    name: "cloudflare-workers-dev-shim",
-    apply: "serve", // dev‑only
-    enforce: "pre",
-    resolveId(id: string) {
-      if (id === "cloudflare:workers") return id; // tell Vite we handled it
-    },
-    load(id: string) {
-      if (id === "cloudflare:workers") {
-        return dedent`
-          import { getCloudflareEnvProxy } from "alchemy/cloudflare";
+  return [
+    {
+      name: "alchemy:workers-dev-shim",
+      apply: "serve", // dev‑only
+      enforce: "pre",
+      resolveId: (id, _importer, options) => {
+        if (id === "cloudflare:workers") {
+          if (!options.ssr) {
+            throw new Error(
+              "cloudflare:workers is not supported in the client",
+            );
+          }
+          return id;
+        }
+      },
+      load: (id) => {
+        if (id === "cloudflare:workers") {
+          return dedent`
+          import { getCloudflareEnvProxy } from "alchemy/cloudflare/env";
           export const env = await getCloudflareEnvProxy(${JSON.stringify(
             options,
           )});
         `;
-      }
+        }
+      },
+    },
+    {
+      name: "alchemy:workers-build-external",
+      config: () => ({
+        build: {
+          rollupOptions: {
+            external: ["cloudflare:workers"],
+          },
+        },
+      }),
```

Comment:
This means you don't have to set it as an external module in your `vite.config.ts`.

We might want to include the other unenv externals? Not sure.

Reply:
Does the cloudflare plugin not do that? Or are we doing this because TSS is not properly using the CF plugin (because of a regression)